spec/tree_spec.rb

```diff
@@ -1,14 +1,97 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
+describe Tree do
```

Comment:
You can probably merge this `describe` block with the one below. Take a look at this too when you get a chance: http://betterspecs.org/

You probably group some of your `it` blocks under `contexts` to improve readability as well.
http://betterspecs.org/#contexts